### PR TITLE
update contract

### PR DESCRIPTION
**Pull request by peggy-gov-bc-ca**
# Description

Update Identifier Contract.
This change will break the Integration system. The dynadapter and icbc adapter needs to change to use the new contract.(the dynadapter and icbc adapter change will be dealt with in following prs.)

## Type of change

Please delete options that are not relevant.

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested? 

Tested locally

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: FAMS-1252, 1254, 1258, 1267

### CODE REVIEW

File: app/SearchApi/SearchAdapter.Sample/SearchRequest/PersonSearchEventSample.cs [hunk 1]
```diff
@@ -28,10 +28,19 @@ public class PersonSample : Person
...
+        public string TypeCode { get; set; }
         public string IssuedBy { get; set; }
+        public string Description { get; set; }
+        public string Notes { get; set; }
+        public IEnumerable<ReferenceDate> ReferenceDates { get; set; }
+    }
+
+    public class ReferenceDateSample : ReferenceDate
+    {
+        public int Index { get; set; }
```
[the remark this answers is not in the document]
Comment (alexjoybc) [reply app/SearchApi/SearchAdapter.Sample/SearchRequest/PersonSearchEventSample.cs]: Index is clear meaning, in a collection we know what an index is, a tag can have many meanings ;-)

File: app/SearchApi/SearchAdapter.Sample/SearchRequest/PersonSearchEventSample.cs [hunk 2]
```diff
@@ -28,10 +28,19 @@ public class PersonSample : Person
...
+        public string Key { get; set; }
```
[the remark this answers is not in the document]
Comment (alexjoybc) [reply app/SearchApi/SearchAdapter.Sample/SearchRequest/PersonSearchEventSample.cs]: We can named it name, I don't see too much difference between the 2, up to the team ;-) what I like with key is that we are in the key value pair concept ;-)

File: app/SearchApi/BcGov.Fams3.SearchApi.Contracts/BcGov.Fams3.SearchApi.Contracts.csproj [hunk 3]
```diff
@@ -2,9 +2,10 @@
 
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
-    <Version>1.0.5</Version>
+    <Version>1.0.6</Version>
```
Comment (alexjoybc) [inline app/SearchApi/BcGov.Fams3.SearchApi.Contracts/BcGov.Fams3.SearchApi.Contracts.csproj]: nice!

File: app/SearchApi/SearchAdapter.Sample/SearchRequest/PersonSearchEventSample.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (rantifamilusi) [reply app/SearchApi/SearchAdapter.Sample/SearchRequest/PersonSearchEventSample.cs]: IEnumerable is already indexed. We are actually tagging to know what to do with data not really indexed. Are we adding index for something else?

File: app/SearchApi/SearchAdapter.Sample/SearchRequest/PersonSearchEventSample.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (rantifamilusi) [reply app/SearchApi/SearchAdapter.Sample/SearchRequest/PersonSearchEventSample.cs]: Can we use IDictionary<key, Value> and loose index. I reckon we need to to add the tag of where to put the data.

File: app/SearchApi/SearchAdapter.Sample/SearchRequest/SearchRequestConsumer.cs [hunk 4]
```diff
@@ -98,10 +98,15 @@ public PersonSearchCompleted BuildFakePersonSearchCompleted(PersonSearchOrdered
...
-                            ExpirationDate = DateTime.Now.AddDays(365),
-                            IssuedBy = "ICBC",
-                            Value = new Random().Next(0, 50000).ToString()
+                            TypeCode = "BCDL",
+                            Description = "Sample Identifier Description",
+                            Notes = "Sample Identifier Notes",
+                            IssuedBy = "BC",
+                            Value = new Random().Next(0, 50000).ToString(),
+                            ReferenceDates = new List<ReferenceDate>(){ 
+                                new ReferenceDateSample(){ Index=0, Key="Effective Date", Value=new DateTime(2019,9,1) },
```
Comment (alexjoybc) [inline app/SearchApi/SearchAdapter.Sample/SearchRequest/SearchRequestConsumer.cs]: Honestly, I like the structure here, it's very self descriptive

**Pull request merged**